exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetFormatPlugin.java

```diff
@@ -250,20 +250,12 @@ private boolean metaDataFileExists(FileSystem fs, FileStatus dir) throws IOExcep
     }
 
     boolean isDirReadable(DrillFileSystem fs, FileStatus dir) {
-      Path p = new Path(dir.getPath(), ParquetFileWriter.PARQUET_METADATA_FILE);
       try {
-        if (fs.exists(p)) {
-          return true;
-        } else {
-
-          if (metaDataFileExists(fs, dir)) {
-            return true;
-          }
-          List<FileStatus> statuses = DrillFileSystemUtil.listFiles(fs, dir.getPath(), false);
-          return !statuses.isEmpty() && super.isFileReadable(fs, statuses.get(0));
-        }
+        // There should be at least one file, which is readable by Drill
+        List<FileStatus> statuses = DrillFileSystemUtil.listFiles(fs, dir.getPath(), false);
+        return !statuses.isEmpty() && super.isFileReadable(fs, statuses.get(0));
```

Comment:
Why did you remove `metaDataFileExists(fs, dir)` check? `DrillFileSystemUtil` won't list dot files.
 Plus it seems you break the assumptions highlighted in `isReadable` method above.

Reply:
I did it on purpose. With the old logic of isDirReadable() method an empty directory, which contains parquet metadata files, will be processes with ParquetGroupScan as a Parquet Table. It leads to obtaining an exception:
https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java#L878

To process such table with SchemalessScan, isReadable method should return false for that case. In other words it shouldn't check availability of metadata cache files, but only really readable files by Drill.

Reply:
1. How we know that we have file in nested directory if filter checks with recursive flag false?
2. The `isReadable` method uses implicit assumption that if metadata file exists, there is some data in folder. Plus comment in this file also  points out that the same logic is used in `isDirReadable`. Since you are removing this logic from `isDirReadable` please make sure the logic is synced between two methods or at least comment is updated to reflect new approach.

Reply:
I have reverted my changes here. SchemalessScan will be created if ParquetGroupScan has empty entries and can't be used. The info that parquet metadata files are invalid will be logged.